src/lib.rs

```diff
@@ -73,8 +73,8 @@ impl<'a, 'b> plugin::Plugin<Request<'a, 'b>> for Raw {
 
         if need_read {
             let max_length = req.get::<persistent::Read<MaxBodyLength>>()
-                .ok().cloned().unwrap_or(DEFAULT_BODY_LIMIT);
-            let body = try!(read_body_as_utf8(req, max_length));
+                .ok().unwrap_or(std::sync::Arc::new(DEFAULT_BODY_LIMIT));
```

Comment:
Can you address your performance enhancement code in a separate change set and merge this PR? Right now it doesn't compile for anybody and renders the project useless. I would like to learn more from your code changes after this merge.